cupy/core/core.pyx

```diff
@@ -1718,6 +1718,14 @@ cdef class ndarray:
         else:
             return NotImplemented
 
+    def __array_function__(self, func, types, args, kwargs):
+        if not hasattr(cupy, func.__name__):
+            return NotImplemented
+        for t in types:
+            if not isinstance(t, ndarray):
```

Comment:
Should be `if not all(t in HANDLED_TYPES for t in types):` where `HANDLED_TYPES` is something like `[np.ndarray, cupy_array]`.

Reply:
Thanks, I fixed.